File: 7_function_overriding.py
class Food:
    cost=None
    type= None
    def __init__(self, cost, type) -> None:
        self.cost= cost
        self.type= type
        
    def display(self):
        print(f"{self.type} of food costs {self.cost} BDT")
        
        
class Fruit(Food):
    def __init__(self, cost, type) :
        super().__init__(cost, type)
        
    def display(self):
        print("This is derived function")
        super().display()
        
cake= Food(30, "Dessert")
apple= Fruit(30, "Healthy")
# cake.display()
apple.display()

File: 7_function_overriding.cpp
// This is related to inheritance

#include <iostream>
#include <bits/stdc++.h>
using namespace std;

class Food
{
    int cost;
    string type;

    public:
    Food(int cost, string type)
    {
        this->cost= cost;
        this->type= type;
    }

    void display(){
        cout<<type<<" type of food costs "<<cost<<" BDT"<<endl;
    }
};

class Fruit: public Food{
    public:
        Fruit(int cost, string type): Food(cost, type){

        }
        void display(){
            cout<<"This is overriden function"<<endl;
            Food::display();
        }
};

int main()
{
    Food cake(30, "Dessert");
    Fruit apple(33, "Healthy");
    cake.display();
    apple.display();
    


    return 0;
}